prov/efa/src/rdm/efa_rdm_ep_fiops.c

```diff
@@ -466,6 +468,21 @@ void efa_rdm_ep_set_use_zcpy_rx(struct efa_rdm_ep *ep)
 		goto out;
 	}
 
+	/* Zero-copy receive requires P2P support. Disable it if any initialized HMEM iface does not support P2P. */
+	for (iface = FI_HMEM_SYSTEM; iface < OFI_HMEM_MAX; ++iface) {
+		hmem_info = &ep->base_ep.domain->hmem_info[iface];
+		if (hmem_info->initialized &&
+		    !hmem_info->p2p_disabled_by_user &&
```

Comment:
If `hmem_info->p2p_disabled_by_user` is disabled by user, we should also disable zcpy, right?

Reply:
I think that's orthogonal - the user can disable P2P for e.g. CUDA, and use zcpy_recv for host memory.

Reply:
Going a step further, why not make `use_zcpy_recv` an attribute of each iface? Otherwise we potentially punish users for simply having *some* HMEM capability for one or more interfaces.

Reply:
That's a valid point but currently the zcpy recv capability is on endpoint, i.e. it's associated with QP instead of hmem iface. In other words, we have to disable zcpy recv for the endpoint if any hmem does not support p2p.

Reply:
`use_zcpy_recv` can only be a global attribute for the ep no matter iface. If we have p2p disabled for cuda while enabled for host, we can only disable zcpy recv